web/src/data/recoil.ts

```diff
@@ -110,18 +129,51 @@ export const vaporConnectionConfig = selector<ConnectionConfig | undefined>({
   },
 });
 
+export const portalConnectionConfig = selector<ConnectionConfig | undefined>({
+  key: "portalConnectionConfig",
+  get: async ({ get }) => {
+    const portalHostnameValue = get(portalHostname);
+    const portalCredentialsValue = get(portalCredentials);
+    const portalDatabaseValue = get(portalCredentials);
+
+    if (portalCredentialsValue) {
+      let decodedCredentials;
+      try {
+        decodedCredentials = atob(portalCredentialsValue);
+      } catch (e) {
+        console.error("error conecting to Portal", { e });
+      }
+      if (portalHostnameValue && decodedCredentials && portalDatabaseValue) {
+        const { username, password } = JSON.parse(decodedCredentials);
+        return {
+          host: "https://" + portalHostnameValue,
+          user: username,
+          password: password,
+          database: portalDatabaseValue,
+        };
+      }
+    }
+  },
+});
+
 export const connectionConfig = selector<ConnectionConfig>({
   key: "connectionConfig",
   get: ({ get }) => {
     const vaporConfig = get(vaporConnectionConfig);
+    const portalConfig = get(portalConnectionConfig);
     if (vaporConfig) {
+      trackAnalyticsEvent("vapor-connection");
       return vaporConfig;
+    } else if (portalConfig) {
+      trackAnalyticsEvent("portal-connection");
```

Comment:
this function could fire many times depending on when the atom is refreshed, if you want to track this you may need to come up with a way to ensure it only happens once